src/plugins/saved_objects/public/save_modal/saved_object_save_modal_dashboard.tsx

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+
+// import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiFormRow, EuiRadio, EuiComboBox, EuiPanel, EuiSpacer } from '@elastic/eui';
+import { SavedObjectDashboard } from '../../../../plugins/dashboard/public';
+import { SavedObjectsClientContract } from '../../../../core/public';
+
+// import { i18n } from '@kbn/i18n';
+import { OnSaveProps, SaveModalState, SavedObjectSaveModal } from '.';
+
+interface SaveModalDocumentInfo {
+  id?: string;
+  title: string;
+  description?: string;
+}
+
+interface DashboardSaveModalProps {
+  documentInfo: SaveModalDocumentInfo;
+  objectType: string;
+  onClose: () => void;
+  onSave: (props: OnSaveProps & { dashboardId: string | null }) => void;
+  savedObjectsClient: SavedObjectsClientContract;
+}
+
+interface SmallDashboard {
+  label: string;
+  value: string;
+}
+
+export function SavedObjectSaveModalDashboard(props: DashboardSaveModalProps) {
+  const [dashboardOption, setDashboardOption] = useState<'new' | 'existing' | null>('existing');
+  const [dashboards, setDashboards] = useState<SmallDashboard[]>([]);
+  const [isLoadingDashboards, setIsLoadingDashboards] = useState(true);
+  const [selectedDashboard, setSelectedDashboard] = useState<SmallDashboard | null>(null);
+
+  const { documentInfo, savedObjectsClient } = props;
+
+  const fetchDashboards = useCallback(
```

Comment:
could probably do with some debouncing here if we're going to call this when a user is typing